src/iterators/linear.rs

```diff
@@ -33,7 +33,7 @@ impl<'a, T: 'a + Counter> Iter<'a, T> {
 impl<'a, T: 'a + Counter> PickyIterator<T> for Iter<'a, T> {
     fn pick(&mut self, index: usize, _: u64) -> bool {
         let val = self.hist.value_for(index);
-        if val >= self.current_step_lowest_value_reporting_level || index == self.hist.last() {
+        if val >= self.current_step_lowest_value_reporting_level || index == self.hist.last_index() {
```

Comment:
I unconsciously expect `last()` to be a _value_, hence `last_index()` to end the uncertainty